### PR TITLE
Add backward single step

### PR DESCRIPTION
**Pull request by feifei-111**
### PR types

Others
### PR changes

Others
### Description

Add backward single step

### CODE REVIEW

File: docs/Tutorial.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+  compare_mode: `mean`
+  single_step: `False`
+}
+[AutoDiff] Assign weight Failed !!!
+
+Error occured between:
+    paddle: `Linear(in_features=100, out_features=100, dtype=float32)` parameter `weight`
+    torch: `Linear(in_features=100, out_features=10, bias=True)` parameter `weight`
+
+Shape of paddle param `weight` and torch param `weight` is not the same. [100, 100] vs [100, 10]
```
Comment (2742195759) [inline docs/Tutorial.md]: 涉及到Weight出问题的，而且需要定位位置的。一定得打印 Path ，而不是Layer的Name，因为name 是唯一的，但是Layer可以有很多个。用户还要数一下。下一个PR修复即可。

File: docs/Tutorial.md [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+Paddle Model
+=========================
+    (net) SimpleLayer
+     |--- (net) Linear    <---  *** HERE ***
+     +--- (net) Linear
+Torch Model
+=========================
+    (net) SimpleModule
+     |--- (net) Linear    <---  *** HERE ***
+     +--- (net) Linear
```
Comment (2742195759) [inline docs/Tutorial.md]: 如果(net)对用户排查问题很重要，最好是说明一下 (net) 标记是什么意思。

File: docs/Tutorial.md [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+         ...
+
+[AutoDiff] FAILED !!!
+```
+
+
+
+### 模型weight/grad对齐失败的报错信息
+
+由于weight/grad对齐信息一般比较多，所以会将信息输入到log文件，并输出log文件路径，请打开打印的文件路径以查看具体信息
```
Comment (2742195759) [inline docs/Tutorial.md]: 提及一下，让用户看下面样例，并在这里说明。

File: docs/Tutorial.md [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+
+## 使用loss & optimizer
+
+### 使用loss
+
+能够向padiff工具传入自定义的loss_fn，并参与对齐。但传入的loss函数有一定限制
+
+须知：
+
+1.   传入的loss_fn是一个可选项，不指定loss_fn时，将使用auto_diff内置的一个fake loss function进行计算
```
Comment (2742195759) [inline docs/Tutorial.md]: 解释一下这个 fake loss function。

File: docs/Tutorial.md [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+
+
+### 使用optimizer
+
+能够向padiff工具传入optimizer，在多step对齐下，将使用optimizer更新模型
+
+须知：
+
+1.   optimizer是可选的，若不传入，padiff并不提供默认的optimzer，将跳过权重更新的步骤
+2.   若需要进行多step对齐，必须传入optimizer
```
Comment (2742195759) [inline docs/Tutorial.md]: 如果没有传入 optimizer，step会被自动重置成为1.

File: docs/Tutorial.md [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+3.   padiff不会检查optimizer内部是否对齐，但在多step下会检查模型权重（受optimizer影响）
+4.   optimizer有两种使用方式：传入一个正常的optimizer（支持clear grad以及step），或者传入一个**无输入的lambda**，并在其中实现自定义操作
```
Comment (2742195759) [inline docs/Tutorial.md]: （传入Torch的Optimzier和Paddle的Optimizer【注意顺序】）

File: docs/Tutorial.md [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+
+## 使用assign_weight
+
+assign_weight 用于复制 torch 模型的权重到 paddle 模型，具体接口参数信息见：[接口信息](Interfaces.md)
+
+assign_weight的逻辑以及报错信息与 auto_diff 开启 auto_weight 选项是一致的，因此可以参考上文
+
+须知：
+
+1.   注意，这个函数不会raise，只会return False并在终端打印信息
```
Comment (2742195759) [inline docs/Tutorial.md]: 如果Assign失败，此函数返回值为False（不会抛出异常）

File: docs/Tutorial.md [hunk 8]
```diff
@@ -0,0 +1,395 @@
...
+2.   由于 padiff 的 api 级别对齐机制目前默认开启，若仅仅需要将assign_weight作为一个辅助其他任务的工具使用，请设置环境变量以关闭api级别对齐机制 `export PADIFF_API_CHECK=OFF`
```
Comment (2742195759) [inline docs/Tutorial.md]: 如果只使用assign weight 接口，不使用 auto_diff 接口，请设置环境变量：`export PADIFF_API_CHECK=OFF`

File: docs/Tutorial.md [hunk 9]
```diff
@@ -0,0 +1,395 @@
...
+
+```py
```
Comment (2742195759) [inline docs/Tutorial.md:389]: import os
os.environment("")

File: padiff/runner/Runner.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+class Runner:
+    def __init__(self, layer, module, loss_fn, opt, layer_map, options):
+        self.trainer = Trainer(layer, module, loss_fn, layer_map, options)
+        self.optimizer_helper = OptimizerHelper(opt, options)
+        self.options = options
+        self.steps = options["steps"]
+        self.layer_map = layer_map
+
+    def do_train(self, paddle_report, torch_report, example_inp):
```
Comment (2742195759) [inline padiff/runner/Runner.py]: do_run

File: padiff/runner/Runner.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        weight_check, grad_check = Checker.weight_grad_check(
+            self.trainer.layer, self.trainer.module, options=self.options, layer_map=self.layer_map
+        )
+        ret = weight_check and grad_check
+        return ret
+
+    def do_optimizer(self):
+        self.optimizer_helper.step()
+
+    def Run(self, example_inp):
```
Comment (2742195759) [inline padiff/runner/Runner.py]: def run

**Review by 2742195759 — DISMISSED** [review]
LGTM

**Review by 2742195759 — APPROVED** [review]
LGTM
